content/knowledge/scalinghistory.md

```diff
@@ -0,0 +1,135 @@
+   As opposed to 51% of all computing power in case of a PoW system.
+   Sustainability, security are achieved in a PoS model like the one in ethereum2.0 which is on its way to be released.
+   PoS ensures more throughput compared to PoW model.
+   However there is a communication overhead which hinders scalability (TODO: Explain more).
+   Also the higher the adaptation, the higher the overall value represented by the overall native cryptocurrency and therefore higher the value that needs to be locked up as staking by the nodes.
+   This means the native currency's full potential isn't utilised as once they are locked, they can no longer be used for any other purposes unless they are unlocked after the locking period expires.
+   Network’s safety is proportional to the value of stake committed.
+
+## 3.1 Sharding and layer 2 solutions
+
+   For solving this scaling issue, 'sharding' is introduced(e.g. In case of [Ethereum2.0](https://www.youtube.com/watch?v=ctzGr58_jeI&t=657s&ab_channel=Finematics).
+   Sharding is a way of spreading the computing and storage workload from a blockchain network, so that each node no longer has to process the entire network's transactional load.
+   Each node only needs to maintain info corresponding to its specific shard or partition.
+   For example, addresses starting with “0x00” will be part of Shard1 and addresses starting with “0x01” will be part of Shard2 etc.
+   There is a mechanism in place for inter-shard communication so that the whole blockchain is still viewable from any node.
+   All the shards will be connected to the “Beacon chain” which will have all the history with it and acts as an orchestrator of the whole network.
+   The consensus between all the shards will be maintained by the Beacon chain.
+   However sharding has its own challenges.
+   Corrupting nodes in a given shard may lead to permanent loss of data.
+   One way to tackle this issue is by randomly assigning a node to a shard and randomly reassigning the node to another shard.
+   Sharding has another problem that the composability aspect of Smart contract compatible blockchain will no longer exist.
+   For example, if a Defi built on top of Ethereum1.0 ecosystem relies on or combines few other projects, then in a sharded system of Ethereum2.0, ideally they should be on the same shard which may not always be the case.
+   If they are not on the same shard then they all needs to be ok with asynchronous updates which is [not ideal](https://www.coindesk.com/tech/2020/10/13/will-a-sharded-ethereum-be-flexible-enough-for-decentralized-finance/).
+
+## 4. Avalanche
+
+   The next level solution for scalability in terms of consensus algorithms is by Avalanche network.
+   In Avalanche, the underlying [sybil control mechanism](https://en.wikipedia.org/wiki/Sybil_attack) is still PoS but there is a different mechanism in place to finalise validating transactions.
```

Comment:
PoS is always just there to prevent sybil attacks. It only defines the set of validators. Then there is a consensus algorithm. Avalanche is essentially still PoS. Might be better to use different terms for the sections:

1. Nakamoto Consensus
2. Federated Byzantine Agreement
3. Byzantine Fault Tolerance
4. Snow Consensus